### PR TITLE
feat(firehose-destinations-alpha): support zero buffering

### LINKED ISSUE #28714: kinesisfirehose-destinations-alpha: support firehose zero buffering
### Describe the feature

Firehose released Firehose zero buffering in [this announcement](https://aws.amazon.com/about-aws/whats-new/2023/12/amazon-kinesis-data-firehose-zero-buffering/) ([documentation](https://docs.aws.amazon.com/firehose/latest/dev/buffering-hints.html)) which allows for setting buffering interval seconds as low as `0` as opposed to the [existing minimum of `60`](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-kinesisfirehose-destinations-alpha/lib/private/helpers.ts#L90) in cdk.

This is a request to allow setting `bufferingInterval` prop as low as `0`.

### Use Case

See released feature

### Proposed Solution

Update the validation to allow for values as low as `0`

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.117.0

### Environment details (OS name and version, etc.)

macOS

### PR DESCRIPTION
**Pull request by humanzz**
support zero buffering ([announcement](https://aws.amazon.com/about-aws/whats-new/2023/12/amazon-kinesis-data-firehose-zero-buffering/), [documentation](https://docs.aws.amazon.com/firehose/latest/dev/buffering-hints.html))

Closes #28714 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by humanzz** [comment]
`Exemption Request`
`Clarification Request`
Package README does not have details as it's deferring to the main firehose's package README

File: packages/@aws-cdk/aws-kinesisfirehose-destinations-alpha/lib/private/helpers.ts [hunk 1]
```diff
@@ -87,8 +87,8 @@ export function createBufferingHints(
   }
 
   const intervalInSeconds = interval?.toSeconds() ?? 300;
-  if (intervalInSeconds < 60 || intervalInSeconds > 900) {
-    throw new Error(`Buffering interval must be between 60 and 900 seconds. Buffering interval provided was ${intervalInSeconds} seconds.`);
+  if (intervalInSeconds > 900) {
```
Comment (scanlonp) [inline packages/@aws-cdk/aws-kinesisfirehose-destinations-alpha/lib/private/helpers.ts:90]:
```suggestion
  if (intervalInSeconds < 0 || intervalInSeconds > 900) {
```
Let's keep the lower bound validation.

Comment (humanzz) [reply packages/@aws-cdk/aws-kinesisfirehose-destinations-alpha/lib/private/helpers.ts:90]: `interval` is a `cdk.Duration` and `cdk.Duration` does not accept negative numbers.
Initially, I wanted to keep the lower bound and when trying to test that, `cdk.Duration` threw an exception about not allowing negative numbers. It was then I decided to drop the lower bound as it's enforced by `cdk.Duration`.

Comment (scanlonp) [reply packages/@aws-cdk/aws-kinesisfirehose-destinations-alpha/lib/private/helpers.ts:90]: Wonderful!

**Comment by humanzz** [comment]
@scanlonp
> Looks good. Just one small change.
> 
> Also, you can add a line in the kinesisfirehose readme about this feature. There is a section on buffering there.

Updated README in [update aws-kinesisfirehose-alpha's README](https://github.com/aws/aws-cdk/pull/28716/commits/75722be8672e405aaeca608e415f178da35d59d4)

Responded to comment abt lower bound inline.

**Review by scanlonp — APPROVED** [review]
Thanks for the addition!

**Pull request merged**